### PR TITLE
Remove jshint and jscs, introduce standard instead

### LINKED ISSUE #54: Switch linters to 'standard'
Since jscs is merging into ESLint, let's switch to ESLint, and while at it, switch to [standard](https://www.npmjs.com/package/standard) to stop maintaining linter configs.

### PR DESCRIPTION
**Pull request by all3fox**
There are no semantic changes in lib/*.js, only syntax changes to fit standard.
Also remove .jshintrc and .jslintrc

Close #54 

This PR changes codestyle dramatically. Also, there is a parallel option of using `eslint-config-standard` config together with `eslint` module. Tell me if you want that instead of `standard` module.

### CODE REVIEW

File: lib/validate.js [hunk 1]
```diff
@@ -1,76 +1,74 @@
...
-	subjectExceptions = /^(fixup|squash)!|^\[[^\]]+\]:/,
-	defaults = {
-		component: true,
-		components: [],
-		limits: {
-			subject: 72,
-			other: 80
-		}
-	};
+var merge = require('mout/object/merge'),
```
Comment (jzaefferer) [inline lib/validate.js]: Let's use this opportunity to get rid of onevar as well. Make each of these use their own `var`

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "install": "node install",
     "uninstall": "node uninstall",
-    "test": "jshint lib/ *.js && jscs lib/ *.js && nodeunit test.js"
+    "test": "standard lib/*.js && nodeunit test.js"
```
Comment (jzaefferer) [inline package.json]: Do we still need the argument passed to `standard`? Let's try if calling just `standard` also works (and fails properly).

**Comment by jzaefferer** [comment]
Thanks, this is looking good. Let me know when you push more commits.

**Comment by all3fox** [comment]
Did what you suggested: `standard` is not called without arguments and checks all the *.js files. So, fixed syntax in all these *.js accordingly. Finally, in all the *.js files changed `onevar` style declarations to several oneliners with their own `var = ` declarations.

The thing that I do not understand is what caused the "this branch has conflicts that must be resolved" message from github (next to the greyed out merge symbol). Tell me if that is something I did wrong.

**Comment by jzaefferer** [comment]
I recently landed two older PRs in master, two of the changes there conflict in this branch. They look easy enough to resolve. A rebase against (upstream) master should fix the warning here.

File: index.js [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-		console.error();
-		console.error( "Commit message was:" );
-		console.error();
-		console.error( chalk.green( message ) );
-		process.exit( 1 );
-	}
-};
+module.exports = function (messageFile) {
+  var message = sanitize(fs.readFileSync(messageFile).toString()),
+    errors = validate(message, options)
```
Comment (jzaefferer) [inline index.js]: Missed a onevar

File: setup.js [hunk 4]
```diff
@@ -1,62 +1,62 @@
...
-var git = path.resolve( root, ".git" );
-var hooks = path.resolve( git, "hooks" );
-
-var rootPackage = require( path.resolve( root, "package" ) );
-if ( rootPackage.commitplease && rootPackage.commitplease.nohook ) {
-	console.error( "package.json indicates to skip hook" );
-	process.exit( 0 );
+;(function avoidSelfInstall () {
+  var pkg = require('./package'),
+    nameRegex = new RegExp('node_modules/' + pkg.name + '$')
```
Comment (jzaefferer) [inline setup.js]: Another onevar

**Comment by all3fox** [comment]
@jzaefferer did the git rebase and fixed the two missed onevars.

**Comment by jzaefferer** [comment]
Nice work!

**Pull request merged**